### DISCUSSION TITLE
does k3s garbage collector collects events?

### DISCUSSION THREAD

**Discussion by migs35323** [k3s-io/k3s#9782] [General]
**Environmental Info:**
K3s Version:  1.27 
(originally 1.23 or 1.24)

Cluster Configuration:
3 masters(servers) + workers(agents), 
(originally 1 master with sqlite as DB)

**Describe the bug:**

I have this cluster for a year and i noticed i still have events from 200+ days ago
i tried to google but i only get info about the containers/images garbage collection..

so i have several questions
1. aren't events supposed to be garbage collected periodically by default?
2. does k3s changes this behavior or makes some custom configuration from upstream kubernetes?
3. choosing to garbage collect event older than <date> is even an option i can change?

**Comment by brandond** [comment]
1. they're not garbage collected, but they do have a TTL
2. no
3. https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/
   > --event-ttl duration     Default: 1h0m0s
   > Amount of time to retain events.

Note that events that continue to occur will be renewed by newer matching events.
